src/swarm/neo/client/RequestOnConn.d

```diff
@@ -329,6 +329,24 @@ public class RequestOnConn: RequestOnConnBase, IRequestOnConn
         Timeout
     }
 
+    /***************************************************************************
+
+        Enumerated states of nodes to indicate their connection statuses.
+
+    ***************************************************************************/
+
+    public enum NodeState
+    {
+        /// The node is connected
+        Connected,
+
+        /// The node does not exist
+        Missing,
+
+        /// The node is disconnected
+        Disconnected
```

Comment:
This did make me wonder what the request handler is recommended to do in the case where the node is known but disconnected. I know that all-nodes requests typically automatically wait for the connection to be reestablished, but this isn't the normal behaviour in single-node requests. I think the norm there is to simply return an error to the user and expect the user to decide what to do (i.e. retry or abandon the request).

No change required here, just noting my thoughts, for fun :)